cmd/swagger/commands/generate/server.go

```diff
@@ -24,55 +24,60 @@ import (
-	WithContext       bool     `long:"with-context" description:"handlers get a context as first arg (deprecated)"`
-	DumpData          bool     `long:"dump-data" description:"when present dumps the json for the template generator instead of generating files"`
-	FlagStrategy      string   `long:"flag-strategy" description:"the strategy to provide flags for the server" default:"go-flags" choice:"go-flags" choice:"pflag"`
-	CompatibilityMode string   `long:"compatibility-mode" description:"the compatibility mode for the tls server" default:"modern" choice:"modern" choice:"intermediate"`
-	SkipValidation    bool     `long:"skip-validation" description:"skips validation of spec prior to generation"`
+	Name                   string   `long:"name" short:"A" description:"the name of the application, defaults to a mangled value of info.title"`
+	Operations             []string `long:"operation" short:"O" description:"specify an operation to include, repeat for multiple"`
+	Tags                   []string `long:"tags" description:"the tags to include, if not specified defaults to all"`
+	Principal              string   `long:"principal" short:"P" description:"the model to use for the security principal"`
+	DefaultScheme          string   `long:"default-scheme" description:"the default scheme for this API" default:"http"`
+	Models                 []string `long:"model" short:"M" description:"specify a model to include, repeat for multiple"`
+	SkipModels             bool     `long:"skip-models" description:"no models will be generated when this flag is specified"`
+	SkipOperations         bool     `long:"skip-operations" description:"no operations will be generated when this flag is specified"`
+	SkipSupport            bool     `long:"skip-support" description:"no supporting files will be generated when this flag is specified"`
+	ExcludeMain            bool     `long:"exclude-main" description:"exclude main function, so just generate the library"`
+	ExcludeSpec            bool     `long:"exclude-spec" description:"don't embed the swagger specification"`
+	WithContext            bool     `long:"with-context" description:"handlers get a context as first arg (deprecated)"`
+	DumpData               bool     `long:"dump-data" description:"when present dumps the json for the template generator instead of generating files"`
+	FlagStrategy           string   `long:"flag-strategy" description:"the strategy to provide flags for the server" default:"go-flags" choice:"go-flags" choice:"pflag"`
+	CompatibilityMode      string   `long:"compatibility-mode" description:"the compatibility mode for the tls server" default:"modern" choice:"modern" choice:"intermediate"`
+	SkipValidation         bool     `long:"skip-validation" description:"skips validation of spec prior to generation"`
+	RegenerateConfigureAPI bool     `long:"regenerate-configureapi" description:"Force regeneration of configureapi.go"`
```

Comment:
Do you really need this? This option is actually already available in the config of the template itself.
You may override it directly without adding this bool.

Reply:
@fredbi what do you mean here?

Reply:
I am adding here a sample config file of mine which overrides things.
In my opinion, you should alter this struct defaults (e.g. before loading the user's own config file)  rather than going with more variables in `GenOpts`.

Note the SkipExists flag and the Source entry referring to the template with implied path (prefixed with asset if we assume this is preloaded).

```yaml
layout:
  application:
    - name: configure
      # Generates configure_api.go. 
      # This is a modified bersion of the standard version, with 
      # automatic registration of API operations handlers and 
      # custom OAuth2 authorization method.
      source: asset:serverConfigureapi
      target: "{{ joinFilePath .Target .ServerPackage }}"
      file_name: "configure_{{ .Name }}.go"
      # Set to false now that our template links to implem
      skip_exists: false
      skip_format: false
      # Generates main for server [disabled]
    - name: main
      source: asset:serverMain
      target: "{{ joinFilePath .Target \"cmd\" (dasherize (pascalize .Name)) }}-server"
      file_name: "main.go"
      # Generates embedded spec
    - name: embedded_spec
      source: asset:swaggerJsonEmbed
      target: "{{ joinFilePath .Target .ServerPackage }}"
      file_name: "embedded_spec.go"
      # Generates server
    - name: server
      source: asset:serverServer
      target: "{{ joinFilePath .Target .ServerPackage }}"
      file_name: "server.go"
      # Generates xxx_api.go (standard template).
    - name: builder
      source: asset:serverBuilder
      target: "{{ joinFilePath .Target .ServerPackage .Package }}"
      file_name: "{{ snakize (pascalize .Name) }}_api.go"
      # Generates doc server. This is a stripped down version of the standard  
      # server, with an interface to allow for composition
    - name: doc
      source: asset:serverDoc
      target: "{{ joinFilePath .Target .ServerPackage }}"
      file_name: "doc.go"
      # Generates common methods for implementation package [specific]
    - name: implementation_common
      source: server/implementation_common.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_common.go"
      skip_exists: false
      skip_format: false
      # Unit test for common methods for implementation package [specific]
    - name: implementation_common_test
      source: server/implementation_common_test.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_common_test.go"
      skip_exists: false
      skip_format: false
      # Generates responders for implementation package [specific]
    - name: implementation_responder
      source: server/implementation_responder.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_responder.go"
      skip_exists: false
      skip_format: false
    - name: implementation_responder_test
      source: server/implementation_responder_test.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_responder_test.go"
      skip_exists: false
      skip_format: false
    - name: implementation_parameter_test
      source: server/implementation_parameter_test.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_parameter_test.go"
      skip_exists: false
      skip_format: false
    - name: implementation_gen_fixtures
      source: server/gen_fixtures.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "gen_fixtures.sh"
      skip_exists: false
      skip_format: true
  models:
    - name: definition
      source: asset:model
      target: "{{ joinFilePath .Target .ModelPackage }}"
      file_name: "{{ (snakize (pascalize .Name)) }}.go"
  operations:
    - name: parameters
      source: asset:serverParameter
      target: "{{ if gt (len .Tags) 0 }}{{ joinFilePath .Target .ServerPackage .APIPackage .Package  }}{{ else }}{{ joinFilePath .Target .ServerPackage .Package  }}{{ end }}"
      file_name: "{{ (snakize (pascalize .Name)) }}_parameters.go"
    - name: responses
      source: asset:serverResponses
      target: "{{ if gt (len .Tags) 0 }}{{ joinFilePath .Target .ServerPackage .APIPackage .Package  }}{{ else }}{{ joinFilePath .Target .ServerPackage .Package  }}{{ end }}"
      file_name: "{{ (snakize (pascalize .Name)) }}_responses.go"
    - name: handler
      source: asset:serverOperation
      target: "{{ if gt (len .Tags) 0 }}{{ joinFilePath .Target .ServerPackage .APIPackage .Package  }}{{ else }}{{ joinFilePath .Target .ServerPackage .Package  }}{{ end }}"
      file_name: "{{ (snakize (pascalize .Name)) }}.go"
  operation_groups:
    # Generates API skeleton methods for implementation package [specific]
    - name: implementation_groups
      source: server/implementation_groups.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ snakize (pascalize .Name) }}.go"
      skip_exists: false
      skip_format: false
    - name: implementation_groups_test
      source: server/implementation_groups_test.gotmpl
      target: "{{ joinFilePath .Target \"implementation\" }}"
      file_name: "implementation_{{ (snakize (pascalize .Name)) }}_test.go"
      skip_exists: false
      skip_format: false
```

Reply:
It looks like a good solution, the only problem is that in the way it works right now, when i load a configuration file it does not append or override only the things that i set, it completely erase the defaults and uses only what is written in the configuration file.
Do you know of  way to set only the things that i write and the rest will be as the default?

Reply:
@fredbi as we spoke on slack, `contribOptionsOverride` is used to override the default flags. this function is called only from one place and allow the user to override with his configuration file